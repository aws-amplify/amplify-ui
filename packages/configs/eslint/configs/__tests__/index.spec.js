const baseConfig = require('..');

describe('base config', () => {
  it('has the expected configuration', () => {
    expect(baseConfig).toMatchSnapshot();
  });
});
