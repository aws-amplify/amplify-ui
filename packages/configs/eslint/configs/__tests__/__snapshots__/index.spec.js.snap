// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`base config has the expected configuration 1`] = `
Object {
  "env": Object {
    "node": true,
  },
  "extends": Array [
    "eslint:recommended",
    "plugin:@typescript-eslint/recommended-requiring-type-checking",
    "prettier",
  ],
  "parser": "@typescript-eslint/parser",
  "parserOptions": Object {
    "ecmaVersion": 12,
    "sourceType": "module",
  },
  "plugins": Array [
    "@typescript-eslint",
  ],
  "rules": Object {
    "@typescript-eslint/explicit-module-boundary-types": 2,
    "@typescript-eslint/member-ordering": "error",
    "@typescript-eslint/no-extra-semi": "error",
    "@typescript-eslint/no-floating-promises": "off",
    "@typescript-eslint/no-unused-expressions": Array [
      "error",
      Object {
        "allowTernary": true,
      },
    ],
    "@typescript-eslint/no-unused-vars": Array [
      "error",
      Object {
        "argsIgnorePattern": "_",
        "varsIgnorePattern": "_",
      },
    ],
    "@typescript-eslint/no-use-before-define": "error",
    "@typescript-eslint/prefer-nullish-coalescing": "error",
    "@typescript-eslint/restrict-template-expressions": "off",
    "@typescript-eslint/unbound-method": "off",
    "comma-dangle": Array [
      "error",
      "only-multiline",
    ],
    "function-paren-newline": "off",
    "generator-star-spacing": "off",
    "global-require": "off",
    "implicit-arrow-linebreak": "off",
    "max-params": 2,
    "no-alert": "error",
    "no-console": "error",
    "no-eval": "error",
    "no-tabs": Array [
      "error",
      Object {
        "allowIndentationTabs": true,
      },
    ],
    "no-unused-vars": "off",
    "prefer-const": "error",
    "prefer-destructuring": Array [
      "error",
      Object {
        "array": false,
        "object": true,
      },
    ],
  },
}
`;
